### ISSUE TITLE
Deleting a like fails

### ISSUE THREAD

**Issue by nekr0z** [idno/Known#1403]
I'm trying to delete a like under one of my older posts (a friend asked for their name to be removed), namely www.evgenykuznetsov.org/2015/diorovo-sunset by clicking a "Delete" link beside it. Instead of the like being promptly deleted, the site asks me to log back in (even if I'm logged in in the first place), and then throws Error 501 on me.

Unfortunately, paid hosted version here, so I can not debug anything or attempt a manual deletion.

**Comment by kylewm** [comment]
Tried deleting a new like on my hosted site at kylewm.withknown.com, and it worked as expected, so there is something more complicated going on here...probably requires someone with access to the hosted log files to debug it.

**Comment by tinokremer** [comment]
Sounds like you are not logged in properly. Do you have access to the admin page?

I have also had a similar issue on hosted withknown when I changed my custom domain. Can you have a look and see what the "Delete" text links to?

**Comment by nekr0z** [comment]
I am not sure what exactly "logged in properly" is supposed to mean. Yes, I have access to admin page, and besides, "Delete" link is not even shown when not logged in in the first place. The link itself looks legit, the only thing that can be suspicious is the fact it's a HTTPS link (and we all know custom domain on a hosted version means really broken HTTPS).

**Comment by tinokremer** [comment]
When you have logged in over https first and you try to access something without https (which I had on my site), you are not logged in on that request. That's because the browser does not send cookies that were set securely to a non-secure site.

Another issue with the same effect was when I setup a custom domain and something went wrong. I was visiting the custom domain, but the "Edit" link and probably also the Delete and all other links were still pointing to the old domain. That also caused the login screen to appear, because no cookies were sent.

If the domain is exactly the same as what you see in the address bar, this is all fine and not the cause of the login screen. I think you need @benwerd or someone who has access to the systems behind WithKnown.

**Comment by nekr0z** [comment]
The link looks allright to me, and I'm pretty positive the post in question and all comments have been created long after the current domain was established for this site. Lack of tools to debug hosted version bothers me, too.

**Comment by kylewm** [comment]
> When you have logged in over https first and you try to access something without https (which I had on my site), you are not logged in on that request. That's because the browser does not send cookies that were set securely to a non-secure site.

This sounds right to me too. You are visiting the page over http://, but the Delete link is https://. Can you visit the https site (and bypass the warning about the certificate not matching *.withknown.com)

https://www.evgenykuznetsov.org/2015/diorovo-sunset

**Comment by nekr0z** [comment]
Appreciate you trying to debug this, Kyle, that was my initial thought, too. Here's what happens:

1. Go to https version discarding the warning. (*Hey, letsencrypt for hosted version, anybody? Just saying...*)
2. Sign in using link in upper right corner.
3. Click "Delete" link beside the desired like.
4. Get a "Welcome back! Sign in!" screen again, sign in again. (*What? I mean, really?*)
5. Get a 501.

In other words, experience is exactly the same.

**Comment by mapkyca** [comment]
4. Yes really :) Allowing session cookies to carry across between http<->https is a security risk, we see this as a session tampering attempt and destroy the session. Known *should* link all urls to the current scheme (http for http, https for https), if this is not happening then that could cause the problem.


Separately, you'll be getting a 501 (most likely, but check the logs) because the POST data will be lost when you get redirected to login, and you then get forwarded to the delete on GET (which is not implemented on that action, so 501). My hunch is that somewhere the session is being destroyed, either a http<->https mismatch, or something in the server config.

**Comment by nekr0z** [comment]
@mapkyca You didn't get the point: I am *already* https and logged in after 1 and 2, so redirecting me to login screen on 4 is certainly a thing that is "not supposed to happen™".

**Comment by tinokremer** [comment]
@nekr0z, I'm sorry to keep asking the same thing over and over again. I have not seen solid confirmation this matches up. Please verify:

- You are on the https version of your site
- You are logged in (hence, without it you would not see the Delete link)
- Use the Right click, Copy link feature to grab the link and paste it here (don't worry, we have no credentials and nobody can use the link without it)

I would like to be absolutely sure the link points to exactly the same domain including https the page is on. It really sounds like that is not the case. If it is, I'm afraid we really do need help from the webhoster to see what is showing up in their logs.

I have tried to reproduce the issue you are having. I did some manual editting in the database on a test site I have. I changed the URL's of the Known site to http instead of https, while I visit it on https. All goes well, but the Delete link points to the http version, exactly as I expect you are seeing as well.

**Comment by nekr0z** [comment]
@tinokremer There you go, see screenshot. Looks like the link is OK: https://www.evgenykuznetsov.org/2015/diorovo-sunset/annotations/09b45c78e4aa99adcb036f1fb1b5084f/delete/
And clicking on this link brings the login screen back, which ot obviously shouldn't.
![w5](https://cloud.githubusercontent.com/assets/1515080/14847366/ebbf0662-0c70-11e6-9b1a-19d13cbf5714.png)

**Comment by tinokremer** [comment]
In that case we ruled out any issues related to the session cookie.

If you open the console (F12), does anything show up there like errors of loading scripts and such? Is that left most addon icon AdBlock? If so, you could try disabling it on your site and see if it helps.

Otherwise hopefully @benwerd could have a look at the logs as it's the WithKnown service.

**Comment by nekr0z** [comment]
@tinokremer Console output looks allright to me (but then again, I'm a psychiatrist, not a coder). Disabling AdBlock doesn't seem to affect anything at all.

**Comment by mapkyca** [comment]
Is this still an issue? Seems something that must have been fixed by now...

**Comment by nekr0z** [comment]
The particular one referred has just worked out for me, so I guess this one is safe to be closed now.
